src/FsCheck/ReflectArbitrary.fs

```diff
@@ -33,22 +33,24 @@ module internal ReflectArbitrary =
        else 
            elems
 
-    let private reflectObj getGenerator =
```

Comment:
Previous reflectObj memoization rarely seemed to succeed. Rewrote in order to cache most of the operation and apply getGenerator on the cached info.